packages/ember-metal/lib/mixin.js

```diff
@@ -257,8 +257,12 @@ function mergeMixins(mixins, m, descs, values, base, keys) {
 
 let IS_BINDING = /^.+Binding$/;
 
-function detectBinding(obj, key, value, m) {
-  if (IS_BINDING.test(key)) {
+export function detectBinding(obj, key, value, m) {
+  if (key.length < 8) { return false; }
+
+  let lastIndex = key.indexOf('Binding', key.length - 7);
```

Comment:
my test is always testing the same string though, i would be curious to see across a wide distributing of strings but interned and not interned.... hmmm